### ISSUE TITLE
[BUG] Jacksonized fails with Cannot construct instance

### ISSUE THREAD

**Issue by cameronbraid** [projectlombok/lombok#2575]
@Jacksonized fails with the following example

```
package demo;

import com.fasterxml.jackson.databind.ObjectMapper;

import lombok.Builder;
import lombok.experimental.SuperBuilder;
import lombok.extern.jackson.Jacksonized;

@Builder
@Jacksonized
public class LombokJacksonized {
  
  private String name;
  
  private Foo foo;

  @SuperBuilder
  @Jacksonized
  public static class Thing {

  }
  
  @SuperBuilder
  @Jacksonized
  public static class Foo extends Thing {

  }
  public static void main(String[] args) throws Exception {
    new ObjectMapper().readValue("{\"foo\":{}}", LombokJacksonized.class);
  }

}
```

```
Exception in thread "main" com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `demo.LombokJacksonized$Foo` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (String)"{"foo":{}}"; line: 1, column: 9] (through reference chain: demo.LombokJacksonized["foo"])
        at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:67)
        at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1611)
        at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:400)
        at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1077)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1320)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:331)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:164)
        at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:542)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:535)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:419)
        at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1310)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:331)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:164)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4482)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3434)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3402)
        at demo.LombokJacksonized.main(LombokJacksonized.java:29)
```

**Comment by janrieke** [comment]
Cannot reproduce with the current edge snapshot and Jackson 2.11.2. Both maven and eclipse work fine.

Do you have the latest lombok snapshot (v1.18.13 "Edgy Guinea Pig" - 2020-08-22)? Which Jackson version are you using?

**Comment by cameronbraid** [comment]
Thanks for taking the time to look into this.

When I run the demo from maven

```
mvn clean compile -U exec:java -Dexec.mainClass="demo.LombokJacksonized"
```

it works, but when running it from vscode it doesn't work.

So I guess vscode is not using the lombok jar defined in pom.xml

**Comment by cameronbraid** [comment]
Yep, verified by copying the lombok-edge.jar into gabrielbb.vscode-lombok-1.0.1/server/lombok.jar and then it works with vscode

**Comment by janrieke** [comment]
Please close this issue if that has solved your problem.

**Comment by mikebm** [comment]
Thank you! This resolved my issue as well. Randomly would fail to construct my instance with errors like:
```(no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)\n at [Source: (org.springframework.util.StreamUtils$NonClosingInputStream); line: 2, column: 2]```

Copying my version of lombok into the extension seems to have fixed it. I went a long time without needing to do this, so no idea why all the sudden I had to.

`cp ~/.m2/repository/org/projectlombok/lombok/1.18.22/lombok-1.18.22.jar ~/.vscode/extensions/gabrielbb.vscode-lombok-1.0.1/server/lombok.jar`
